controllers/prep_handlers.go

```diff
@@ -125,89 +108,67 @@ func readPrecachingList(imageListFile, clusterRegistry, seedRegistry string) (im
 	return imageList, nil
 }
 
-func updateProgressFile(path, progress string) error {
-	return os.WriteFile(common.PathOutsideChroot(path), []byte(progress), 0o700)
-}
-
-func (r *ImageBasedUpgradeReconciler) launchPrecaching(ctx context.Context, imageListFile, progressfile string, ibu *lcav1alpha1.ImageBasedUpgrade) (result ctrl.Result, err error) {
+func (r *ImageBasedUpgradeReconciler) launchPrecaching(ctx context.Context, imageListFile string, ibu *lcav1alpha1.ImageBasedUpgrade) (bool, error) {
 	clusterRegistry, err := clusterinfo.NewClusterInfoClient(r.Client).GetReleaseRegistry(ctx)
 	if err != nil {
 		r.Log.Error(err, "Failed to get cluster registry")
-		return
+		return false, err
 	}
 
 	seedInfo, err := clusterinfo.ReadClusterInfoFromFile(
 		common.PathOutsideChroot(getSeedManifestPath(getDesiredStaterootName(ibu))))
 	if err != nil {
 		r.Log.Error(err, "Failed to read seed info")
-		return
+		return false, err
 	}
 
 	imageList, err := readPrecachingList(imageListFile, clusterRegistry, seedInfo.ReleaseRegistry)
 	if err != nil {
 		err = fmt.Errorf("failed to read pre-caching image file: %s, %w", common.PathOutsideChroot(imageListFile), err)
-		return
+		return false, err
 	}
 
 	// Create pre-cache config using default values
 	config := precache.NewConfig(imageList)
 	err = r.Precache.CreateJob(ctx, config)
 	if err != nil {
 		r.Log.Error(err, "Failed to create precaching job")
-		if err = updateProgressFile(progressfile, "Failed"); err != nil {
-			r.Log.Error(err, "Failed to update progress file for precaching")
-			return
-		}
-		return
-	}
-	if err = updateProgressFile(progressfile, "precaching-in-progress"); err != nil {
-		r.Log.Error(err, "Failed to update progress file for precaching")
-		return
+		return false, err
 	}
-	r.Log.Info("Precaching progress status updated to in-progress")
 
-	result = requeueWithShortInterval()
-	return
+	return true, nil
 }
 
-func (r *ImageBasedUpgradeReconciler) queryPrecachingStatus(ctx context.Context, progressfile string) (result ctrl.Result, err error) {
-	// fetch precaching status
-	status, err := r.Precache.QueryJobStatus(ctx)
+func (r *ImageBasedUpgradeReconciler) queryPrecachingStatus(ctx context.Context) (status *precache.Status, err error) {
+	status, err = r.Precache.QueryJobStatus(ctx)
 	if err != nil {
-		r.Log.Error(err, "Failed to get precaching job status")
+		r.Log.Info("Failed to get precaching job status")
```

Comment:
Do we really want to skip printing the actual error?

Reply:
There will be redundant logging of the error since the error is logged in line 465 (`r.Log.Error(err, "Prep stage failed with error")`) when the prepStageWorker function returns the error. Hence I removed the log.Error here

Reply:
I think we may want to create a separate PR in which we review redundant & duplicated logs in other places in an attempt to clean-up the LCA logs.